python/lsst/daf/butler/transfers/_context.py

```diff
@@ -79,8 +81,35 @@ def __init__(self, registry: Registry, datastore: Datastore, backend: RepoExport
         self._directory = directory
         self._transfer = transfer
         self._records: Dict[DimensionElement, Dict[DataCoordinate, DimensionRecord]] = defaultdict(dict)
-        self._dataset_ids = set()
-        self._datasets: Dict[Tuple[DatasetType, str], List[FileDataset]] = defaultdict(list)
+        self._dataset_ids: Set[int] = set()
+        self._datasets: Dict[DatasetType, Dict[str, List[FileDataset]]] \
+            = defaultdict(lambda: defaultdict(list))
+        self._collections: Dict[str, CollectionRecord] = {}
+
+    def saveCollection(self, name: str) -> None:
+        """Export the given collection.
+
+        Parameters
+        ----------
+        name: `str`
+            Name of the collection.
+
+        Notes
+        -----
+        `~CollectionType.RUN` collections are also exported automatically when
+        any dataset referencing them is exported.  They may also be explicitly
+        exported this method to export the collection with no datasets.
+        Duplicate exports of collections are ignored.
+
+        Exporting a `~CollectionType.TAGGED` or `~CollectionType.CALIBRATION`
+        collection will cause its associations with exported datasets to also
+        be exported, but it does not export those datasets automatically.
```

Comment:
so you can end up with an empty calibration collection after import? Or a broken one because the references exist but the datasets don't?

Reply:
Empty is indeed possible (I think that's unavoidable if a partial export of a collection is allowed).  Broken should not be possible.